tests/s2n_test.h

```diff
@@ -200,7 +200,16 @@ int test_count;
 /* for use with S2N_RESULT */
 #define EXPECT_OK( function_call )  EXPECT_TRUE( s2n_result_is_ok(function_call) )
 
-#define EXPECT_BYTEARRAY_EQUAL( p1, p2, l ) EXPECT_EQUAL( memcmp( (p1), (p2), (l) ), 0 )
+#define EXPECT_BYTEARRAY_EQUAL( p1, p2, l ) \
+    do { \
+        /* Memcmp is annotated with a non-null attribute. The if statement prevents \
+         * calling memcmp with NULL arguments when the length is zero. \
+         */ \
```

Comment:
https://github.com/aws/s2n-tls/pull/4722#discussion_r1728059863 Don't assume the GCC nonnull attribute, and if you could, you'd want to specify that you're referring to a GCC / compiler attribute. "attribute" is too vague.

Reply:
I remove the comment.